packages/ai-providers/server-ai-langchain/src/ldai_langchain/langchain_provider.py

```diff
@@ -0,0 +1,273 @@
+                data={},
+                raw_response='',
+                metrics=LDAIMetrics(
+                    success=False,
+                    usage=TokenUsage(total=0, input=0, output=0),
+                ),
+            )
+
+    def get_chat_model(self) -> BaseChatModel:
+        """
+        Get the underlying LangChain model instance.
+
+        :return: The underlying BaseChatModel
+        """
+        return self._llm
+
+    # =============================================================================
+    # STATIC UTILITY METHODS
+    # =============================================================================
+
+    @staticmethod
+    def map_provider(ld_provider_name: str) -> str:
+        """
+        Map LaunchDarkly provider names to LangChain provider names.
+
+        This method enables seamless integration between LaunchDarkly's standardized
+        provider naming and LangChain's naming conventions.
+
+        :param ld_provider_name: LaunchDarkly provider name
+        :return: LangChain-compatible provider name
+        """
+        lowercased_name = ld_provider_name.lower()
+
+        mapping: Dict[str, str] = {
+            'gemini': 'google-genai',
+        }
+
+        return mapping.get(lowercased_name, lowercased_name)
+
+    @staticmethod
+    def get_ai_metrics_from_response(response: BaseMessage) -> LDAIMetrics:
+        """
+        Get AI metrics from a LangChain provider response.
+
+        This method extracts token usage information and success status from LangChain responses
+        and returns a LaunchDarkly AIMetrics object.
+
+        :param response: The response from the LangChain model
+        :return: LDAIMetrics with success status and token usage
+
+        Example:
+            # Use with tracker.track_metrics_of for automatic tracking
+            response = await tracker.track_metrics_of(
+                lambda: llm.ainvoke(messages),
+                LangChainProvider.get_ai_metrics_from_response
+            )
+        """
+        # Extract token usage if available
+        usage: Optional[TokenUsage] = None
+        if hasattr(response, 'response_metadata') and response.response_metadata:
+            token_usage = response.response_metadata.get('tokenUsage') or response.response_metadata.get('token_usage')
+            if token_usage:
+                usage = TokenUsage(
+                    total=token_usage.get('totalTokens', 0) or token_usage.get('total_tokens', 0),
+                    input=token_usage.get('promptTokens', 0) or token_usage.get('prompt_tokens', 0),
+                    output=token_usage.get('completionTokens', 0) or token_usage.get('completion_tokens', 0),
+                )
+
+        # LangChain responses that complete successfully are considered successful by default
+        return LDAIMetrics(success=True, usage=usage)
+
+    @staticmethod
+    def create_ai_metrics(langchain_response: BaseMessage) -> LDAIMetrics:
```

Comment:
There is no need to include the deprecated method since this has never been released. It was left in js-core because it was already in use when we decided to rename it.